File: fixtures/scale/lua/large/checks/check_driftstream.lua
local driftstream = require("ciphers.driftstream")
local driftstream_spec = require("specs.driftstream_spec")
local corpus_stream = require("corpus.corpus_stream")

local function check_driftstream()
  local spec = driftstream_spec.get()
  for _, text in ipairs(corpus_stream.corpus_stream()) do
    local sealed = driftstream.driftstream_encrypt(text, spec.key)
    local opened = driftstream.driftstream_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "stream"
end

return { check_driftstream = check_driftstream }

File: fixtures/scale/javascript/large/src/checks/checkDriftstream.js
import driftstream from '../ciphers/driftstream.js';
import driftstreamSpecMod from '../specs/driftstreamSpec.js';
import corpusStreamMod from '../corpus/corpusStream.js';

function checkDriftstream() {
  const spec = driftstreamSpecMod.driftstreamSpec();
  for (const text of corpusStreamMod.corpusStream()) {
    const sealed = driftstream.driftstreamEncrypt(text, spec.key);
    const opened = driftstream.driftstreamDecrypt(sealed, spec.key);
    if (opened !== text) return false;
  }
  return spec.category === 'stream';
}

export default { checkDriftstream };
